### PR TITLE
Feature/make non spa proxy launch setting

### PR DESCRIPTION
**Pull request by gogeterobert**
Added back the proxy, but created a new profile that is run by default when you `dotnet run` that runs without proxy

### CODE REVIEW

File: src/Web/Web.csproj [hunk 1]
```diff
@@ -38,7 +35,12 @@
 		<Exec WorkingDirectory="$(SpaRoot)\" Command="npm install" />
 	</Target>
 
-	<Target Name="PublishRunWebpack" BeforeTargets="Build" DependsOnTargets="NpmInstall">
+	<Target Name="NpmRunBuild" AfterTargets="NpmInstall">
+		<Exec WorkingDirectory="$(SpaRoot)\" Command="npm run build"/>
+	</Target>
+
+	<Target Name="PublishRunWebpack" AfterTargets="ComputeFilesToPublish">
+		<Exec WorkingDirectory="$(SpaRoot)\" Command="npm install" />
```
Comment (bacongobbler) [inline src/Web/Web.csproj:43]: Don't we want to have this depend on the `NpmInstall` target? That way we aren't duplicating logic. For example this target does not check for the presence of `node` on the system whereas the `NpmInstall` target does.

File: src/Web/Web.csproj [hunk 2]
```diff
@@ -23,11 +25,6 @@
 		<None Include="$(SpaRoot)\**" Exclude="$(SpaRoot)\node_modules\**" />
 	</ItemGroup>
 
-	<ItemGroup>
-		<!-- Extends watching group to include Angular app changes -->
```
Comment (bacongobbler) [inline src/Web/Web.csproj:27]: Why was this removed?

File: src/Web/Web.csproj (hunk 1 above)
Comment (gogeterobert) [reply src/Web/Web.csproj:43]: The problem with depending on the `NpmInstall` target is that it only triggers when there is no node_modules, which is not enough. If we update / add a npm library it will not trigger and it will not install the incoming changes, resulting in an error. As an alternative, we can remove the condition and make this depend on npminstall.

Comment (bacongobbler) [reply src/Web/Web.csproj:43]: We could extend the condition to check if package.json has been changed. That would handle both cases:

1. We want to run `npm install` any time new packages are introduced
2. We do NOT want to run `npm install` if we already have everything installed.

```
<Target Name="NpmInstall" Inputs="$(SpaRoot)\package.json" Outputs="node_modules/.install-stamp">
    ...
    <!-- Write the stamp file, so incremental builds work -->
    <Touch Files="node_modules/.install-stamp" AlwaysCreate="true" />
</Target>
```

File: src/Web/Web.csproj (hunk 2 above)
Comment (bacongobbler) [reply src/Web/Web.csproj:27]: Is there a way to keep this check in place when we're using the `no-proxy` launch target? That way users running `dotnet watch` without Visual Studio will allow their environment to reload on any changes.

File: src/Web/Web.csproj (hunk 1 above)
Comment (gogeterobert) [reply src/Web/Web.csproj:43]: Ok, will look into doing that

**Comment by bacongobbler** [comment]
Merging for now. Let's fix up in a follow-up PR.
